src/wallet/mod.rs

```diff
@@ -316,6 +316,51 @@ where
         }
     }
 
+    /// Ensures that there are at least `max_addresses` or [CACHE_ADDR_BATCH_SIZE] addresses
+    /// cached in the database if the descriptor is derivable, or 1 address if it is not.
+    /// Will return `Ok(true)` if there are new addresses generated (either external or internal),
+    /// and `Ok(false)` if at the required addresses are already cached. This function is used internally
```

Comment:
I think this is what you meant.
```suggestion
    /// and `Ok(false)` if all the required addresses are already cached. This function is used internally
```

Reply:
Good catch! no typechecker for bad english 😅. will post an update